### PR TITLE
Attestation Pruning

### PR DESCRIPTION
**Pull request by GregTheGreek**
This pr introduces attestation pruning before the opPool starts up. It removes attestations that are outside of the min and max inclusion delay

### CODE REVIEW

File: packages/lodestar/src/opPool/modules/attestation.ts [hunk 1]
```diff
@@ -13,4 +13,13 @@ export class AttestationOperations extends OperationsModule<Attestation> {
       return isValidAttestationSlot(config, attestationSlot, state.slot);
     });
   }
+
+  public async pruneInvalid(state: BeaconState, config: IBeaconConfig): Promise<void> {
+    const attestations: Attestation[] = await this.getAll();
+    const invalidAttestations: Attestation[] = attestations.filter((a: Attestation) => {
+      const attestationSlot: Slot = getAttestationDataSlot(config, state, a.data);
+      return !isValidAttestationSlot(config, attestationSlot, state.slot)
```
Comment (mpetrunic) [inline packages/lodestar/src/opPool/modules/attestation.ts]: isnt this removing attestations that are waiting for inclusion delay?

Comment (mpetrunic) [reply packages/lodestar/src/opPool/modules/attestation.ts]: and not removing ones that aren't targeting current or previous epoch?

**Review by mpetrunic — CHANGES_REQUESTED** [review]

**Review by mpetrunic — APPROVED** [review]

**Review by mpetrunic — APPROVED** [review]

**Comment by wemeetagain** [comment]
I think we can close this in favor of the interop branch.
https://github.com/ChainSafe/lodestar/pull/439/files#diff-3608c095dc65cd197d70c53caeb0f243R24

**Comment by GregTheGreek** [comment]
agreed

**Comment by mpetrunic** [comment]
@GregTheGreek @wemeetagain This PR is trying to merge into interop, did we ever pushed this functionality to interop branch?

**Comment by GregTheGreek** [comment]
@mpetrunic there is a function called [removeOld()](https://github.com/ChainSafe/lodestar/pull/439/files#diff-3608c095dc65cd197d70c53caeb0f243R24) that got pushed :)

**Pull request closed without merging**